### PR TITLE
Add CDL Checkout Page

### LINKED ISSUE #4030: If an item is eligible for CDL and the user is logged in, they should get a copyright statement and the option to check out the item.
Copyright statement:

> This Item may be protected by third-party copyright and/or related intellectual property rights. It is provided by Princeton University Library on a non-commercial basis for your personal fair-use academic and educational research purposes only. For other uses, you must obtain permission from the rights-holder. In addition, you may not make copies of the Item, display it online (e.g., on the internet), or distribute it to anyone else, including friends, colleagues, or classmates. By continuing to use this digital resource you are acknowledging and agree to comply with these terms of use. For additional details, see the following Copyright Notice.
> 
> NOTICE WARNING CONCERNING COPYRIGHT RESTRICTIONS
> 
> The copyright law of the United States (title 17, United States Code) governs the making of photocopies or other reproductions of copyrighted material.
> 
> Under certain conditions specified in the law, libraries and archives are authorized to furnish a photocopy or other reproduction. One of these specific conditions is that the photocopy or reproduction is not to be “used for any purpose other than private study, scholarship, or research.” If a user makes a request for, or later uses, a photocopy or reproduction for purposes in excess of “fair use,” that user may be liable for copyright infringement.
>  
> This institution reserves the right to refuse to accept a copying order if, in its judgment, fulfillment of the order would involve violation of copyright law.

If it's available have a big visible "Check Out" button.

If it's not available, say "This item is currently checked out. The estimated wait time is (x) hours."

Checking out an item should create a `CDL::ChargedItem` instance for a free item in bibdata and attach it to its manager, then reload the page (which should now load the viewer.)

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #4030

### CODE REVIEW

**Comment by hackmastera** [comment]
@tpendragon can you add screenshots to this PR?

File: spec/controllers/viewer_controller_spec.rb [hunk 1]
```diff
@@ -51,11 +51,75 @@
...
+
+          expect(response).to be_successful
+          expect(response.body).to have_content "This Item may be protected by third-party copyright and/or related intellectual property rights."
+          expect(response.body).to have_button "Check Out", disabled: true
+          expect(response.body).to have_content "This item is currently checked out. The estimated wait time is about 3 hours."
+        end
+      end
+    end
+    context "when the user is logged in and the resource is private and CDL ineligible" do
+      it "redirects back to the viewer" do
```
Comment (hackmastera) [inline spec/controllers/viewer_controller_spec.rb:114]: The viewer doesn't show it though, right? If it's private?

**Comment by tpendragon** [comment]
Screen Shot 2020-07-07 at 1 27 36 PM
Screen Shot 2020-07-07 at 1 41 04 PM
Screen Shot 2020-07-07 at 1 41 50 PM

File: spec/controllers/viewer_controller_spec.rb (hunk 1 above)
Comment (tpendragon) [reply spec/controllers/viewer_controller_spec.rb:114]: Right. It'll just show a blank page. There's no valid way to get to this state.

**Comment by tpendragon** [comment]
Screen Shot 2020-07-07 at 1 45 11 PM

**Comment by tpendragon** [comment]
@hackmastera Added some screenshots from OL's perspective.

**Comment by hackmastera** [comment]
Thank you! I think we should add some padding around the words; it seems like they come up really close to the edge of the iframe.

File: spec/controllers/viewer_controller_spec.rb (hunk 1 above)
Comment (hackmastera) [reply spec/controllers/viewer_controller_spec.rb:114]: Can we add this as a comment to the test and/or the code?

**Comment by tpendragon** [comment]
Sure, I added a bit of padding.
Screen Shot 2020-07-08 at 10 08 07 AM

@hackmastera re-review?

**Comment by tpendragon** [comment]
Uploaded agreed upon changes.
Screen Shot 2020-07-08 at 11 20 05 AM

@hackmastera can you approve?

**Pull request merged**